api/base/filters.py

```diff
@@ -3,12 +3,26 @@
 import operator
 
 from modularodm import Q
+from modularodm.query.queryset import BaseQuerySet
 from rest_framework.filters import OrderingFilter
 from rest_framework import serializers as ser
 
 from api.base.exceptions import InvalidFilterError
 from api.base import utils
 
+def get_multisort_fn(fields):
+    fields = list(fields)
+    def sort_fn(a, b):
+        while fields:
+            field = fields.pop(0)
+            a_field = getattr(a, field)
+            b_field = getattr(b, field)
+            if a_field > b_field:
+                return 1
+            elif a_field < b_field:
+                return -1
+        return 0
+    return sort_fn
```

Comment:
Writing your own cmp function is very slow according to the python 2.7 docs. It says each element will be touched twice with a cmp function while standard sorted and sorted with a key function only touches each once. Using a key function on the other hand is very fast. There is an attrgetter function built into the operator library. 

`return sorted(queryset, key=operator.attrgetter(*ordering))` seems to kill about 12 LOC with the same functionality and according to the docs more speed.